radixdlt/src/main/java/com/radixdlt/properties/PersistedProperties.java

```diff
@@ -0,0 +1,194 @@
+			this.properties.store(propertiesOutput, "");
+		}
+	}
+
+	/**
+	 * Returns the property if there is a property matching the key, otherwise {@code null}.
+	 *
+	 * @param key the property key
+	 * @return the property if there is a property matching the key, otherwise {@code null}
+	 */
+	public String get(String key) {
+		return this.properties.getProperty(key);
+	}
+
+	/**
+	 * Returns a property value as an {@code int}.
+	 *
+	 * @param key the property key
+	 * @param defaultValue the default value returned if no value is set
+	 * @return either the property value, or {@code defaultValue} if no value set
+	 */
+	public int get(String key, int defaultValue) {
+		String value = get(key);
+		return value == null ? defaultValue : Integer.parseInt(value);
+	}
+
+	/**
+	 * Returns a property value as a {@code long}.
+	 *
+	 * @param key the property key
+	 * @param defaultValue the default value returned if no value is set
+	 * @return either the property value, or {@code defaultValue} if no value set
+	 */
+	public long get(String key, long defaultValue) {
+		String value = get(key);
+		return value == null ? defaultValue : Long.parseLong(value);
+	}
+
+	/**
+	 * Returns a property value as a {@code boolean}.
+	 * Note that the strings "true", ignoring case, and correctly formatted non-zero
+	 * integers are {@code true} values, all others are {@code false}.
+	 *
+	 * @param key the property key
+	 * @param defaultValue the default value returned if no value is set
+	 * @return either the property value, or {@code defaultValue} if no value set
+	 */
+	public boolean get(String key, boolean defaultValue) {
+		String value = get(key);
+		return value == null ? defaultValue : parseBoolean(value);
+	}
+
+	/**
+	 * Returns a property value as a {@code String}.
+	 *
+	 * @param key the property key
+	 * @param defaultValue the default value returned if no value is set
+	 * @return either the property value, or {@code defaultValue} if no value set
+	 */
+	public String get(String key, String defaultValue) {
+		String value = get(key);
+		return value == null ? defaultValue : value;
+	}
+
+	/**
+	 * Sets a property value.
+	 *
+	 * @param key the property name
+	 * @param value the property value
+	 */
+	public void set(String key, Object value) {
+		this.properties.setProperty(key, value.toString());
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(this.properties);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null || !this.getClass().equals(obj.getClass())) {
+			return false;
+		}
+		PersistedProperties other = (PersistedProperties) obj;
+		return Objects.equals(this.properties, other.properties);
+	}
+
+	@Override
+	public String toString() {
+		return Objects.toString(this.properties);
+	}
+
+	private boolean parseBoolean(String value) {
+		if (value.trim().equalsIgnoreCase("true")) {
+			return true;
+		}
+		// Try as 0 = false, not-0 = true
```

Comment:
I'm not very fond of this algorithm for determining truthiness. Do we actually use numbers as boolean fields anywhere in our configs? If not, can we get rid of this?

Reply:
Yes, this is for maintaining backward compatibility.